tests/rptest/services/utils.py

```diff
@@ -19,7 +23,7 @@ def __str__(self):
         example = next(iter(example_lines))
 
         summary = ','.join([
-            f'{i[0].account.hostname}({len(i[1])})'
+            f'{self._get_hostname(i[0])}({len(i[1])})'
```

Comment:
_get_hostname is a static method but is being called as `self._get_hostname` - how does that work?

Reply:
It is a virtual overload for statically defined method. Virtual overloaded method will be called only if child class context is present, but not if called directly.

Example:
```
class A():
    @staticmethod
    def one():
         print("one")

    def myprint(self):
        self.one()

class B(A):
    @staticmethod
    def one():
        print("two")

a = A()
b = B()
# class context is a
a.myprint() - > "one"
a.one() -> "one"
# class context is b, and myprint has access to virtual link via 'self'
b.myprint() -> "two"
# class context is b, but since it is static, virtual link non-existent and original parent method is called
b.one() -> "one"
# context is b, and whole virtual table is accessible as it is called as 'cls.method()'
B.one() -> "two"
```

Reply:
And I am actually wrong and the virtual table in python3 is local for current context and overrides work as they should
```
a.myprint()
one
a.one()
one
b.myprint()
two
b.one()
two
B.one()
two
```

Reply:
Right, this was a brain fart on my part, of course you can call any static method using `self` or `ClassName` and it works (using the actual type of the object, hence following the usual MRO semantics for objects of that type). The only magic with regular methods (non-static) is that self is additionally passed as the first parameter, but lookup is not affected (unlike say C++ which is probably where this brain fart came from).

Reply:
Well, my brain goes overdrive when I write such things in C++. In Python, it is in relaxed mode.